extractor/src/extract/extractor.py

```diff
@@ -244,15 +243,16 @@ def split_dataframe(df, analysis):
         prev_pk_val = None
         acc = defaultdict(list)
         for row in df:
-            if acc and row[pk_col] != prev_pk_val:
+            pk_ind = row.keys().index(pk_col)
+            if acc and row[pk_ind] != prev_pk_val:
```

Comment:
awful (but I agree, not a lot of options). Can you add either a comment or type hints so we understand what `row` is ?